client/chat/route.tsx

```diff
@@ -1,30 +1,59 @@
-import React from 'react'
-import { useRoute } from 'wouter'
-import { makeSbChannelId } from '../../common/chat'
+import React, { Suspense } from 'react'
+import { Route, RouteProps, Switch } from 'wouter'
+import { makeSbChannelId, SbChannelId } from '../../common/chat'
+import { hasAnyPermission } from '../admin/admin-permissions'
 import { replace } from '../navigation/routing'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppSelector } from '../redux-hooks'
 import { ConnectedChatChannel } from './channel'
+import { ChannelList } from './channel-list'
+import { CreateChannel } from './create-channel'
 
-export function ChannelRouteComponent(props: { params: any }) {
-  const [matches, params] = useRoute('/chat/:channelId/:channelName')
+const getLoadableChatAdminComponent = (isAdmin: boolean) =>
```

Comment:
Can you please just write this inline? Routes can just have children instead of passing a `component` and this makes things needlessly complex

Reply:
I'll also just note that `React.lazy` creates a *component*, so what you have done here is made a thing that creates a brand new component type every time these routes are rendered, so please never write code like this

https://react.dev/reference/react/lazy#my-lazy-components-state-gets-reset-unexpectedly